### PR TITLE
[Issue #551 FixUp] Add read permission to server-key.pem

### PR DESCRIPTION
**Pull request by nccu-ben**
Signed-off-by: Ben Ho <pohan.ho@gmail.com>

### CODE REVIEW

**Comment by nccu-ben** [comment]
@romfreiman @KiranSukhavasi 
can you review my pull request? I appreciate your help.

File: component-samples/demo/scripts/web_csr_req.sh [hunk 1]
```diff
@@ -22,3 +22,5 @@ openssl x509 -x509toreq -in webServer.pem -out server.req -signkey server-key.pe
 #comment out following line if signing with external CA
 openssl x509 -req -days 12775 -in server.req -CA ca-cert.pem -CAkey caKey.pem -CAcreateserial -out server-cert.pem -extfile ./web-server.conf -extensions v3_req  
 
+chmod 644 server-key.pem
```
Comment (DukeDavis12) [inline component-samples/demo/scripts/web_csr_req.sh:25]: We generally don't run chmod in a script.
https://stackoverflow.com/questions/68063830/including-chmod-in-shell-script-will-it-work-for-everyone
Least Privilege Principle.

We ask customers to `chmod -R 644 secrets/ service.env`

**Comment by DukeDavis12** [comment]
Can you contribute for #580 ?

**Comment by nccu-ben** [comment]
yes, I can.

File: component-samples/demo/scripts/server.req [hunk 2]
```diff
@@ -0,0 +1,17 @@
+-----BEGIN CERTIFICATE REQUEST-----
+MIICujCCAaICAQAwdTELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAk9SMRIwEAYDVQQH
+DAlIaWxsc2Jvcm8xEDAOBgNVBAoMB0xGIEVkZ2UxFDASBgNVBAsMC0ZETyBwcm9q
+ZWN0MR0wGwYDVQQDDBRob3N0LmRvY2tlci5pbnRlcm5hbDCCASIwDQYJKoZIhvcN
+AQEBBQADggEPADCCAQoCggEBAPGUDrGohyn0djN1+RcW4+T0Kp27VsyYCXtoyGEn
+JR4lgXas3A5Lp31cDd8PHdpApkr/iBHQtUn2eut26yX+Fax4zmqC1pWT+oAWaMiF
```
Comment (DukeDavis12) [inline component-samples/demo/scripts/server.req:6]: Can you remove the .req files ?

File: component-samples/demo/scripts/ca-cert.srl [hunk 3]
```diff
@@ -0,0 +1 @@
+6AB57CCB438066D978DFFC4196F01B289C1A0E19
```
Comment (DukeDavis12) [inline component-samples/demo/scripts/ca-cert.srl:1]: Please remove ca-cert.srl

Comment (DukeDavis12) [reply component-samples/demo/scripts/ca-cert.srl:1]: Please don't include any secrets in the PR

Comment (nccu-ben) [reply component-samples/demo/scripts/ca-cert.srl]: Thank you for your advice.
another change is spell error on Readme

Comment (nccu-ben) [reply component-samples/demo/scripts/ca-cert.srl]: I passed the local test, are there any details I need to pay attention to?
截圖 2023-04-10 下午3 54 51

File: component-samples/demo/scripts/client.req [hunk 4]
```diff
@@ -0,0 +1,17 @@
+-----BEGIN CERTIFICATE REQUEST-----
+MIICrTCCAZUCAQAwaDEQMA4GA1UEAwwHYXBpVXNlcjEUMBIGA1UECwwLRkRPIHBy
+b2plY3QxEDAOBgNVBAoMB0xGIEVkZ2UxEjAQBgNVBAcMCUhpbGxzYm9ybzELMAkG
+A1UECAwCT1IxCzAJBgNVBAYTAlVTMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIB
+CgKCAQEAkkVLS06VmJNK5rHh1TjvF4WiLpp1hOenXmHBOICE0RT8uw5J7Lz0xoQ1
+wj9XedChpYY75e4KMCTqykSae+bh5TQLERvo3j9bggIgeTCbXSp4R3aekbq+vQL9
+ByDQBDOAPHtOcrkJikTzTWA4mMcohrbKThrYBk/Q6ba9RobK/tb2DGN5/me4ck6s
+f2tqaJMcObGYWA4N9N7NY7bSPxaACu+aUEXjlYepTg5seB9mxUJOwSy09eRFxDUb
```
Comment (DukeDavis12) [inline component-samples/demo/scripts/client.req:8]: Please remove credentials files like client.req,db_password.txt

Comment (DukeDavis12) [reply component-samples/demo/scripts/client.req:8]: server.req

File: component-samples/demo/scripts/service.env [hunk 5]
```diff
@@ -0,0 +1,9 @@
+## environment variables
+db_user=root
+api_password=default
+db_password=db_password.txt
+encrypt_password=fMl89mpGaqgPt7J
```
Comment (DukeDavis12) [inline component-samples/demo/scripts/service.env:5]: Remove this file.

File: component-samples/demo/scripts/db_password.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+UKCZaHfQWCBfz9kV
```
Comment (DukeDavis12) [inline component-samples/demo/scripts/db_password.txt:1]: Remove this file.

File: component-samples/demo/scripts/secrets/db_password.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+UKCZaHfQWCBfz9kV
```
Comment (DukeDavis12) [inline component-samples/demo/scripts/secrets/db_password.txt:1]: Remove this file.

**Comment by DukeDavis12** [comment]
One more suggestion; I had a talk with the shell expert and he is fine with your `chmod 644` usage.

**Comment by nccu-ben** [comment]
Thank you! This is my new PR.
https://github.com/secure-device-onboard/pri-fidoiot/pull/585

Davis Benny ***@***.***> 於 2023年4月11日 週二 上午12:51寫道：

**Pull request closed without merging**